### PR TITLE
chore: disable tests on Python3.6

### PR DESCRIPTION
**Pull request by dee-me-tree-or-love**
# What

Since the update of `ubuntu-latest` image in [setup-python](https://github.com/actions/setup-python) GitHub Action to `Ubuntu 22.04`, Python 3.6 is no longer available by default. https://github.com/pypiserver/pypiserver/actions/runs/3861663821/jobs/6683883640#step:3:10. See related issue: https://github.com/actions/setup-python/issues/561.  

Since Python 3.6 is declared "end of life" and is planned to be deprecated for `pypiserver 2.0.0`, I believe it's worth disabling these tests. Related to https://github.com/pypiserver/pypiserver/issues/107 and the discussion with @cclauss in https://github.com/pypiserver/pypiserver/pull/447.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10", "pypy3.9"]  # "3.11-dev"
+        python-version: ["3.7", "3.8", "3.9", "3.10", "pypy3.9"]  # "3.11-dev"
```
Comment (cclauss) [inline .github/workflows/ci.yml:21]: Add 3.11?
Add 3.12-dev?

Comment (dee-me-tree-or-love) [reply .github/workflows/ci.yml:21]: True, thanks! :) Since I'm on it already, that's a great idea 👍

**Comment by cclauss** [comment]
I would comment out `Python 3.12-dev`

`check` is failing because there is a new version of `psf/black` that removes some blank lines in Python code.

File: requirements/dev.pip [hunk 2]
```diff
@@ -6,6 +6,6 @@
 -r exe.pip
 -r test.pip
 
-black
+black==22.12.*
```
Comment (cclauss) [inline requirements/dev.pip:9]: #474 instead?

**Comment by dee-me-tree-or-love** [comment]
Hey @cclauss, thank a lot for your participation and suggestions! That was very useful. And I see the new PR about black - that's great, I will review it shortly! For now I will proceed with merging this PR to enable going further, thanks for your help 👍

**Pull request merged**